app/appsettings.cpp

```diff
@@ -216,6 +218,101 @@ void AppSettings::setLegacyFolderMigrated( bool hasBeenMigrated )
   emit legacyFolderMigratedChanged( mLegacyFolderMigrated );
 }
 
+const QString &AppSettings::activePositionProviderId() const
+{
+  return mActivePositionProviderId;
+}
+
+void AppSettings::setActivePositionProviderId( const QString &id )
+{
+  if ( mActivePositionProviderId == id )
+    return;
+
+  mActivePositionProviderId = id;
+  setValue( QStringLiteral( "activePositionProviderId" ), id );
+  emit activePositionProviderIdChanged( mActivePositionProviderId );
+}
+
+QVariantList AppSettings::savedPositionProviders() const
+{
+  QSettings settings;
+  QVariantList providers;
+
+  int size = settings.beginReadArray( mPositionProvidersArrayGroupName );
+
+  for ( int i = 0; i < size; i++ )
+  {
+    settings.setArrayIndex( i );
+    QString name = settings.value( "providerName" ).toString();
+    QString address = settings.value( "providerAddress" ).toString();
+    providers.append( { name, address } );
+  }
+
+  settings.endArray();
+
+  return providers;
+}
+
+void AppSettings::savePositionProvider( const QString &name, const QString &address )
+{
+  QSettings settings;
+  QVariantList providers = savedPositionProviders();
+
+  if ( !providers.isEmpty() )
+  {
+    settings.remove( mPositionProvidersArrayGroupName );
+  }
+
+  // ignore if we already have such address saved
+  for ( int i = 0; i < providers.count(); i++ )
+  {
+    if ( providers[i].toList()[1] == address )
+      return;
+  }
+
+  providers.append( { name, address } );
+
+  writePositionProvidersArray( providers );
+  emit savedPositionProvidersChanged( providers );
+}
+
+void AppSettings::removePositionProvider( const QString &address )
+{
+  QSettings settings;
+  QVariantList providers = savedPositionProviders();
+
+  if ( !providers.isEmpty() )
+  {
+    settings.remove( mPositionProvidersArrayGroupName );
+  }
+
+  // do we already have such address saved?
+  for ( int i = 0; i < providers.count(); i++ )
+  {
+    if ( providers[i].toList()[1] == address )
+    {
+      providers.removeAt( i );
+      break;
+    }
+  }
+
+  writePositionProvidersArray( providers );
+  emit savedPositionProvidersChanged( providers );
+}
+
+void AppSettings::writePositionProvidersArray( QVariantList providers )
+{
+  QSettings settings;
+  settings.beginWriteArray( mPositionProvidersArrayGroupName );
+
+  for ( int i = 0; i < providers.count(); i++ )
```

Comment:
I think I always use ++i , but maybe it is the same in this context (?)

Reply:
It makes no difference in for loop context :)